diskcache/core.py

```diff
@@ -2450,3 +2454,32 @@ def reset(self, key, value=ENOVAL, update=True):
 
         setattr(self, key, value)
         return value
+
+if hasattr(os, 'register_at_fork'):
+    _lock = threading.RLock()
+
+    def _acquireLock():
+        global _lock
+        try:
+            _lock.acquire()
+        except BaseException:
+            _lock.release()
+            raise
+
+    def _releaseLock():
+        global _lock
+        _lock.release()
+
+    def _after_at_fork_child_reinit_locks():
+        global _lock
+        _lock = threading.RLock()
```

Comment:
Why re-init the lock in the child? I would expect the child to already have a copy of the lock.